core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/BCFile.java

```diff
@@ -403,20 +355,13 @@ public void close() throws IOException {
           long offsetIndexMeta = out.position();
           metaIndex.write(out);
 
-          if (cryptoParams.getCipherSuite() == null || cryptoParams.getCipherSuite()
-              .equals(Property.CRYPTO_CIPHER_SUITE.getDefaultValue())) {
-            out.writeLong(offsetIndexMeta);
-            API_VERSION_1.write(out);
-          } else {
-            long offsetCryptoParameters = out.position();
-            cryptoParams.write(out);
-
-            // Meta Index, crypto params offsets and the trailing section are written out directly.
-            out.writeLong(offsetIndexMeta);
-            out.writeLong(offsetCryptoParameters);
-            API_VERSION.write(out);
-          }
+          long offsetCryptoParameter = out.position();
+          String es = this.encryptionStrategy.getClass().getName();
+          out.writeUTF(es);
```

Comment:
I don't think the overall encryption strategy should be serialized into the files. The pluggable encryption strategy could change, and the current strategy should be required to be able to interpret anything previously written. So, I don't think it's necessary to serialize the name of the strategy here... but perhaps we should permit it some opportunity write some magic values/parameters that it (or its future replacement, if swapped out) could recognize?

Reply:
The `table.file.crypto.strategy` and `tserver.wal.crypto.strategy` were something I conjured up on a whim.  Not sure if there are pitfalls to this approach.  Could have system level config for selecting crypto like the following.

```
crypto.strategy.s201805.class=com.foo.Best
crypto.strategy.s201805.props.mykey=XU24GNYXE74VSGJM
crypto.strategy.s201903.class=com.foo.Better
crypto.strategy.s201903.props.thekey=56GU5ET6TQKZZCKP
crypto.current.strategy=s201903
```

Reply:
I like these are good ideas.  I wrote the class name because it was easiest and wasn't sure how to handle changing encryption strategies. 

@ctubbsii do @keith-turner 's ideas match up with your thoughts?

Reply:
Yeah, I think using an identifier that isn't (necessarily) the class name is best for this kind of thing.

Reply:
An alternative approach is to not have an id in the config. Also do not have the class name in the config.  Instead the strategy can provide a versioning string at write time that is persisted in the rfile.  At read time this versioning string is provided back to the strategy.  This empowers the strategy to handle versioning in whatever what it likes.

Reply:
If there is no encryption strategy class in the config or the file then how do you know what class to load?

Reply:
That should be determined by an overall crypto module, which is capable of supplying an appropriate decrypter for a given id.

Reply:
Oh my bad I mean to write : An alternative approach is to not have an id in the config. Also do not have the class name in the *rfile*.